pyrometheus/chem_expr.py

```diff
@@ -241,14 +242,34 @@ def troe_falloff_expr(react: ct.Reaction, t):
     :returns: The Troe falloff center expression for reaction *react* in terms of the
         temperature *t* as a :class:`pymbolic.primitives.Expression`
     """
-    troe_params = react.falloff.parameters
+    if react.uses_legacy:
+        from warnings import warn
+        warn("New Cantera implementation retrieves 'FalloffRate' objects "
+             "using the rate 'property'", DeprecationWarning)
```

Comment:
- What is "new Cantera"? (Be precise: version numbers and such.)
- Good messages are actionable: what should the user do to fix this?
- Deprecations typically have a fuse ("when will this stop working?"). Specify it, if it exists, or specify that it doesn't exist.
- If the message is a full sentence, it needs punctuation.

Reply:
(I don't recall a situation where I found a warning or error message to have too much explanation or context.)